### ISSUE TITLE
[ScopedElementsMixin] Element scoping not working as expected when we have Angular Components with LitComponents inside

### ISSUE THREAD

**Issue by msbasanth** [open-wc/open-wc#2205]
## Expected behavior

I have an angular app where different UI parts are from different teams and they are using their own version of LitElement UI components. With scoped elements I should be able to use multiple versions of same ui components.

## Actual Behavior

When I create container component with LitElement  [ScopedElementsMixin] it works fine.
```
Main Angular AppComponent
	--> ui-part-lit-element [ScopedElementsMixin]
		--> LitElement1
		--> LitElement2
		--> LitElement3
```
But we have mostly this structure in our application where we have many angular components made out of LitElement ui components.

```
Main AppComponent
	--> ui-part-lit-element [ScopedElementsMixin]
		--> LitElement1
		--> LitElement2
		--> AngularComponent1
			--> LitElement1.1
			--> LitElement1.2
```

## Additional context
For making sure that ScopedElementsMixin avoids collision I used a top level container. This is the topmost ScopedElementsMixin container which I used for keeping the UI part from particular team. As they also have angular component I tried adding it to the topmost component.

```
class ScopeUiPart1 extends ScopedElementsMixin(LitElement) {
    static get scopedElements() {
        return {
            'access-bar': AccessBar
        };
    }

    static get styles() {
        return css`
    :host {
      width: 100%;
      height: 100%;
      display: block;
    }
    `;
    }

    render() {
        return html`
        <inner-angular-component></inner-angular-component>
    	<access-bar native="true"></access-bar>
    `;
    }
}

````

When I checked DOM [ScopedElementsMixin] it tried to scope even the angular component and it won't render.
![image](https://user-images.githubusercontent.com/17777813/117776368-1e975600-b259-11eb-8dfd-587dfbbac53e.png)

Is this the right way to combine UI from a particular team in need of versioning.

Or does it mean that we need a topmost LitElement container for each UI part. i.e. in this case I need a LitElement container inside inner-angular-component, so that the ui components inside that angular compponent also gets unique tag name.

Thanks
Basanth

**Comment by manolakis** [comment]
Hello @msbasanth,

sorry for the late response. Unfortunatelly, I don't think `ScopedElementsMixin`  version 1 could work in this way, as it would try to scope the angular component thinking it could be a lazy definition. A workaround could be wrap the angular element inside a LitElement, this way the lit element would be scoped instead of the angular one.

However, `ScopedElementsMixin` version 2, as is using the scoped registry polyfill (https://www.npmjs.com/package/@webcomponents/scoped-custom-element-registry), the tag names are not going to be scoped, so I think it should work.

**Comment by msbasanth** [comment]
Thanks for the reply, we will look into & follow scoped registry polyfill.

**Comment by artzfly** [comment]
kkhk

**Comment by msbasanth** [comment]
@manolakis Yes with ScopedElements_v2.x tag names are no longer being rewritten with a hash. That means we can still use custom element tags as CSS Selector.

I tested different scenarios with LitElement components using ScopedElementsMixin **inside Angular App**. 

For an Angular application, we needed a host element that creates a local custom element registry,
**Approach 1: Creating a host element using Lit + ScopedElementsMixin2.0 which internally handles CustomElementRegistry**
![image](https://user-images.githubusercontent.com/17777813/200725455-2b7c2475-5a85-41e0-b4eb-9d05fee6048f.png)
**Approach 2: Create a host element using HTMLElement by manually creating and passing CustomElementRegistry to attachShadow**
![image](https://user-images.githubusercontent.com/17777813/200725612-fa69a026-a2bf-4ffc-a533-6b7d817aaebf.png)

In both the approaches observations are almost same. Sharing the observations here,

| Sl.No. |  Scenario| <div style="width:500px">Hierarchy</div> | Observations |
|--|--|--|--|
| 1  | Angular Component inside shadowdom of LitElement | ![image](https://user-images.githubusercontent.com/17777813/200721861-4ba38fd8-9a3d-4430-9f5a-a4ae206d357b.png) | **Not working** as Lit is unable to render Angular Component as expected. |
| 2  | Angular Component inside a slot of LitElement  | ![image](https://user-images.githubusercontent.com/17777813/200723195-65ee8995-a75b-4eb6-9685-97386c150431.png)| **Working**, Lit is able to render Angular Component with content projection. |
| 3 | Added an Angular Component with LitElement inside into a slot inside Host LitElement | ![image](https://user-images.githubusercontent.com/17777813/200723303-4b143b45-51b5-451e-86af-2e305bf2b72c.png) |  Angular Component working, Lit custom element inside Angular Component not rendered (LitElement2). Looks like local custom element registry is not available here|
| 4  | **Angular Element**  inside shadowdom of LitElement | ![image](https://user-images.githubusercontent.com/17777813/200726591-bfb4a7e7-7b7b-40ff-b396-e853fb4bb32c.png) | Unable to pass CustomElementRegistry into AngularElement so that they use local registry instead of global registry <br> https://github.com/angular/angular/issues/44110 |
|  |  |  |  |

**Observation Summary**

- Angular Components we may not be able to use inside host element which is needed for scoping.
- Need **Angular Elements** as they are standard custom element created from Angular Components. Which Lit renderer can render just like any other custom element.
- We basically need custom elements having a "local" CustomElementRegistry and children of that customElement using that local version instead of the global window.customElements one.
- Now @angular/elements do not support passing local CustomElementRegistry so that they can use it to instantiate the elements instead of document.createElement (global registry). 

So I think until we get support from **Angular Elements** we do not have a cleaner way of using scoped elements 2.0 with Angular. Or I am missing something here?
